### PR TITLE
Documentation fixes and CI update

### PR DESCRIPTION
**Pull request by palmskog**
I removed the obsolete `description` file and made some documentation improvements and updates in `meta.yml`. Then I regenerated all boilerplate from the latest templates. Also includes an optional Dune build.

### CODE REVIEW

File: coq-coqoban.opam
Comment (erikmd) [inline coq-coqoban.opam:21]: Wouldn't it be better to keep the `| (= "dev")` constraint?

Indeed, it seemed to me it was necessary for testing with `coqorg/coq:dev`;

but I'm a bit puzzled to see that the corresponding job was green… do you have an explanation?

**Review by erikmd — APPROVED** [review]
Thanks @palmskog! LGTM.

Just waiting for your feedback on my review comment before merging (and doing a release :)

BTW, I'd like to suggest changing the version naming convention for coqoban.

→ Currently, it is like this:
[coq-coqoban](http://coq.io/opam/coq-coqoban.8.11.0.html).{8.5.0, 8.6.0, 8.7.0, 8.8.0, 8.9.0, 8.10.0, 8.11.0}

but given that the project is "cross-compatible" with many coq versions, I believe this naming scheme is not relevant anymore…

File: coq-coqoban.opam
Comment (palmskog) [reply coq-coqoban.opam:21]: Since opam is based on `dpkg` style version comparisons, you need an 8.XX upper bound for the `| (= "dev")` part to be meaningful - otherwise it's redundant. When we only have `>= "8.10"`, `coq.dev` is still allowed:
```shell
dpkg --compare-versions 8.10 lt dev && echo true
true
```

**Comment by palmskog** [comment]
> but given that the project is "cross-compatible" with many coq versions, I believe this naming scheme is not relevant anymore…

Maintainers are of course free to switch version scheme, but since this is a low-intensity project, I think the `8.XY.Z` scheme is the simplest one to follow that will be backwards-compatible. Whenever you want to make a release, just check what the Coq `8.XY` release is the latest stable one, and use `8.XY.0`. Users won't really care much as long as it builds/installs, is my experience.

File: coq-coqoban.opam
Comment (erikmd) [reply coq-coqoban.opam:21]: OK! I think I figured out what happens there:

it appears `"dev" >= "8.10"`, so the or clause can be removed.

But that wouldn't be the case if, e.g., `coq-coqoban` required `{(>= "8.10") & (< "8.14~")}`.

Comment (erikmd) [reply coq-coqoban.opam:21]: @palmskog, sorry for some reason GitHub didn't show me your previous message before I replied myself 😅

**Pull request merged**
